portal_pages/templatetags/portal_extras.py

```diff
@@ -11,6 +11,66 @@ def make_unique(value, unique_var): # Only one argument.
     else:
         return value
 
+@register.filter
+def remove_from_string(value, value_to_remove): 
+    """Remove a value from a list of values, ie. in order to unselect a Country from a filter of Countries"""
+    if value.find("," + value_to_remove) >= 0:
+        value = value.replace("," + value_to_remove, "")
+    elif value.find(value_to_remove) >= 0:
+        value = value.replace(value_to_remove, "")
+
+    return value
+
+@register.inclusion_tag('portal_pages/tags/filter_list.html', takes_context=True)
+def display_filter_list(context, filter_type, tags_header, items, request_list):
+
+    context.request.tags_header = tags_header # ie. countries
+    context.request.tags_header_title = tags_header.title() # ie. Countries
+    
+    request_vars = context.request.GET.get(filter_type,"")
+
+    search_query = ""
+    if context.request.POST.get("search", ""):
+        search_query =  "&search=" + context.request.POST.get("search", "")
+    elif context.request.GET.get("search", ""):
+        search_query = "&search=" + context.request.GET.get("search", "")
+
+    addl_filters_string = ""
+    for request_item in request_list.split(","):
+        addl_filters_string = addl_filters_string + "&" + request_item + "=" + context.request.GET.get(request_item, "")
+
+    if request_vars: # Collapse the div if no filters are currently active for this filter
+        context.request.collapse_state = "uncollapsed"
+    else:
+        context.request.collapse_state = "collapsed"
+
+    context.request.link_items = []
+    for item in items:
+        if item.name in request_vars:
+            item_class = "active"
+            # Remove this active item in the list to toggle filter off
+            request_vars_list = request_vars.lstrip(",").split(",") # Strip out any leading comma
+            request_vars_list.remove(item.name)
+            item_href="?" + filter_type + "=" + ",".join(request_vars_list)
+        else:
+            item_class = ""
+            request_vars_string = (request_vars + "," + item.name).lstrip(",") # Strip out any leading comma
+            item_href="?" + filter_type + "=" + request_vars_string
+
+        # Append the search query to the string
+        if search_query:
+            item_href = item_href + search_query
+
+        # Append all the other filter variables to the URL string
+        if addl_filters_string:
+            item_href = item_href + addl_filters_string
+
+        context.request.link_items.append({"class": item_class, "href": item_href, "name": item.name})
+    
+    return {
+        'request': context['request'],
+    }
```

Comment:
There is no need to use an existing context object and annotate it to pass back in to render the inclusion tag's template. In this case it would make more sense to return a dict from this function that has "tags_header", "collapse_state", and "link_items" rather than annotating request with these things. Simplifies both the code and the template...